### DISCUSSION TITLE
Page.show_pdf_page() function alternative/work around

### DISCUSSION THREAD

**Discussion by TimHuynh0905** [pymupdf/PyMuPDF#1582] [Q&A]
Hi there,

I am trying to find drawings inside a specific area (clip) of a page in a similar manner to Page.get_text(). However, both Page.get_drawings() and Page.get_cdrawings() do not provide that functionality out of the box. 

I wonder if there is a way to work around this issue. If possible, why don't you add a clip parameter inside the Page.get_drawings() and Page.get_cdrawings()?

One way I considered using to work around this problem is to use Document.show_pdf_page(..., clip = <area of original page>). But as being said in the documentation, all resources "will be imported ...  even if they are not contained in the visible area given by clip." Thus, Page.get_cdrawings() still won't work the way I want.

References:
- https://pymupdf.readthedocs.io/en/latest/page.html#Page.get_cdrawings
- https://pymupdf.readthedocs.io/en/latest/page.html#Page.show_pdf_page

Thank you for your help!

Cheers.

**Comment by JorjMcKie** [comment]
The simplest thing to do surely is
```python
paths = [p for p in page.get_drawings() if p["rect"] in clip]
```
Done.

**Comment by JorjMcKie** [comment]
If you however had in mind to include the subset of a drawing that is partly contained in the clip (and some parts outside) ...
This is not possible to implement.
So please let me know your intention.

**Comment by TimHuynh0905** [comment]
Thank you so much for your responses! 

In my current project, my intention is to extract an area out of a page and export it as a stand-alone pdf file. I want to do some analyses specifically on this area only. Apparently, the show_pdf_page() function also imports all resources outside of the visible clip area, thus get_drawing() outputs things I don't need. Do you have any recommendation to work around it?

Thank you!

**Comment by JorjMcKie** [comment]
If you set a page's `/CropBox` to the area of interest, you will at least only get stuff **_displayed_** from this area.

Other than mentioned above I have no advice. This is BTW the reason why I detest introducing a clip parameter: (1) the above one-line is too trivial to make a big deal out of it, and (2) it **_does not work_** like the clip parameter of other methods (pixmaps, text extractions,...): there, a precise cut-out of the clip happens, whereas a drawing cannot be partially contained - it is either in or out completely.

Well, it seems that if you make an SVG image from a page with modified cropbox, the results may better. Haven't checked how far this algorithm goes though.
At least once you **_have_** the SVG, you could convert it back to a PDF ...
Tell me your findings please.
